### DISCUSSION TITLE
Graphql-helix vs graphl-ez.

### DISCUSSION THREAD

**Discussion by spa5k** [PabloSzx/graphql-ez#230] [Q&A] [answered]
Looking into the docs, graphql ez looks like it's combination of graphql-helix + envelop combined into one, is there some other difference? like in terms of performance or something?

A comparison table in doc will do great.

**Comment by benbender** [comment]
@Unbuttun You are right. It is a combination of `graphql-helix` and `envelop`. But it also adds a nice, clean api to adapt in different contexts/servers and a plugin-api, with quite some plugins already there, which rescues you from all the fiddling and boilerplate to make those common tasks work in your environment.

So, if you want to, you can think about it as the plugin-system for the app-part, where envelop does the plugin-system for the graphql-part.

And about performance: Technically it should be nearly the same cost/benefit as if you would do those setups by hand... But as you'll find quite some options for caching in here, it could be easily a gain in net-cost, I would say :)

**Comment by PabloSzx** [comment, marked as answer]
thank you @benbender for the description, very well explained.

@Unbuttun I don't think there is a "difference" with graphql-helix + envelop, graphql-ez uses both, if you want differences and compare with alternatives, it would be with apollo server, express-graphql or Mercurius for example, and graphql-ez philosophy is to be relatively unopinionated and give maximum customization and composability leveraging Envelop at it's maximum, this project was born thanks to Envelop, and graphql-ez is also meant to work as an engine for more opinionated setups, like the next version of graphql-yoga
